Keyboard/Keyboard.swift

```diff
@@ -33,6 +35,33 @@ class Keyboard: UIView {
         handleButtonTap(sender: sender)
         delegate?.handleStateChange()
     }
+    
+    override init(frame: CGRect) {
+        super.init(frame: frame)
+        let contrastBarTranslation = CGFloat(contrastBarSize)
+        //contrast bar
+        addSubview(contrastBar)
+        contrastBar.backgroundColor = .black
+        contrastBar.translatesAutoresizingMaskIntoConstraints = false
+        addConstraint(NSLayoutConstraint(item: contrastBar, attribute: .left, relatedBy: .equal, toItem: self, attribute: .left, multiplier: 1, constant: 0))
+        addConstraint(NSLayoutConstraint(item: contrastBar, attribute: .top, relatedBy: .equal, toItem: self, attribute: .top, multiplier: 1, constant: 0))
+        addConstraint(NSLayoutConstraint(item: contrastBar, attribute: .bottom, relatedBy: .equal, toItem: self, attribute: .top, multiplier: 1, constant: contrastBarTranslation))
+        addConstraint(NSLayoutConstraint(item: contrastBar, attribute: .right, relatedBy: .equal, toItem: self, attribute: .right, multiplier: 1, constant: 0))
+    }
+    
+    // used at the end of sublass init functions to properly add the contrast bar
+    // does not handle any adjusting of text or other ui element, simply adds the
+    // contrastView as a subView to all given views, so that touches will still be read
+    func addContrastBar(views: [UIView]) {
```

Comment:
I'd like to avoid doing this stuff if possible.
Take a look at [this function](https://developer.apple.com/reference/uikit/uiview/1622469-hittest) to see if we can change this.